tests/test_cli.py

```diff
@@ -22,3 +31,140 @@ def test_operation_result_repr():
+            input_json.flush()
+            input_json.seek(0)
+            result = WeaverClient().execute("fake_process", input_json.name, url="http://fake.domain.com")
+    assert result is mock_result
+    assert len(inputs) == 1
+    assert inputs[0] == {"input1": "data"}
+
+
+@pytest.mark.cli
+def test_parse_inputs_with_media_type():
+    inputs = []
+    mock_result = OperationResult(True, code=500)
+
+    def parsed_inputs(_inputs, *_, **__):
+        inputs.append(_inputs)
+        return mock_result
+
+    # catch and ignore returned error since we generate it
+    # voluntarily with following mock to stop processing
+    def no_error_cli(*args):
+        weaver_cli(*args)
+        return 0
+
+    with mock.patch("weaver.cli.WeaverClient._update_files", side_effect=parsed_inputs):
+        with tempfile.NamedTemporaryFile(mode="w", suffix=".yml") as input_yaml:
+            yaml.safe_dump({"info": {"data": "yaml"}}, input_yaml)
+            input_yaml.flush()
+            input_yaml.seek(0)
+
+            run_command([
+                # weaver
+                "execute",
+                # different media-type than YAML on purpose to ensure parsing uses provided value, and not extension
+                "-I", f"input:File={input_yaml.name}@mediaType={ContentType.APP_CWL}",
+                "-p", "fake_process",
+                "-u", "http://fake.domain.com",
+                "-q",  # since CLI fails purposely, avoid logging errors which would be confusing if debugging logs
+            ], entrypoint=no_error_cli)
+
+    assert len(inputs) == 1
+    assert inputs[0] == {
+        "input": {
+            "href": input_yaml.name,  # normally, local file would be uploaded to vault, but we skipped it with mock
+            "format": {"mediaType": ContentType.APP_CWL}
+        }
+    }
+
+    inputs.clear()
+    schema_json = "http://schema.org/random.json"
+    schema_xml = "http://schema.org/other.xml"
+    with mock.patch("weaver.cli.WeaverClient._update_files", side_effect=parsed_inputs):
+        with ExitStack() as stack:
+            input_yaml = stack.enter_context(tempfile.NamedTemporaryFile(mode="w", suffix=".yml"))
+            yaml.safe_dump({"info": {"data": "yaml"}}, input_yaml)
+            input_yaml.flush()
+            input_yaml.seek(0)
+            input_json = stack.enter_context(tempfile.NamedTemporaryFile(mode="w", suffix=".json"))
+            json.dump({"info": {"data": "json"}}, input_json)
+            input_json.flush()
+            input_json.seek(0)
+            input_xml = stack.enter_context(tempfile.NamedTemporaryFile(mode="w", suffix=".xml"))
+            input_xml.write("<xml>data</xml>")
+            input_xml.flush()
+            input_xml.seek(0)
+            input_tif = stack.enter_context(tempfile.NamedTemporaryFile(mode="wb", suffix=".tif"))
+            input_tif.write(base64.b64encode("012345".encode("utf-8")))
+            input_tif.flush()
+            input_tif.seek(0)
```

Comment:
Knowing you, I find it weird that you didn't wrap those 3 lines in a function 😛.